DeviceConnectionAndReliabilityReadme.md

```diff
@@ -0,0 +1,63 @@
+# Azure IoT Device Client .NET SDK
+
+## Device connection and messaging reliability
+
+### Overview
+
+In this document you will find information about:
+
+- The connection authentication and renewal methods.
+- The reconnection logic and retry policies.
+- The timeout controls.
+
+### Connection authentication
+
+Authentication can be done using either:
+
+- [SAS tokens](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-security#security-tokens) - [Sample usage](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs#L102)
+- [x509 certificates](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-security#supported-x509-certificates) - [Sample usage](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/X509DeviceCertWithChainSample/Program.cs#L43)
+- [Device Provisioning Service](https://docs.microsoft.com/azure/iot-dps) - [Sample usage](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/PnpDeviceSamples/Thermostat/Program.cs#L90)
+
+When using SAS tokens, authentication can be done by:
+
+- Providing the shared access key and letting the SDK create the SAS tokens by using one of the [CreateFromConnectionString](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/DeviceClient.cs#L121) options on the DeviceClient or an authenticationMethod other than DeviceAuthenticationWithTokenRefresh.
+
+    If you choose this option, the SDK will create the SAS tokens and renew them before expiry. The default values for time-to-live and renewal buffer can be changed using the [ClientOptions](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/ClientOptions.cs) properties.
+    - SasTokenTimeToLive: The suggested time-to-live value for tokens generated for SAS authenticated clients. Default value is 60 minutes.
```

Comment:
Some thoughts I have here are to augment the readme with links to the doc comments along side--or in some cases instead of--the code implementation in GitHub. The implementation is great for us and other folks trying to improve the client but others will see this doc as a getting started guide and might be more inclined to explore the API this way.

Reply:
Sure sounds like a good idea. I like the idea of pointing to MSDN docs better than pointing to our code. Is that what you mean?

Reply:
Correct.